### DISCUSSION TITLE
so where *is* the dictionary?

### DISCUSSION THREAD

**Discussion by anarcat** [Automattic/harper#170] [Q&A] [answered]
So I'm reviewing this long document that has a metric ton of acronyms and things, and I want to build a dictionary so I don't ever have to spell out Prometheus and Grafana to Harper ever again.

The harper-ls docs say the dictionary is in `$XDG_CONFIG_HOME/harper-ls/`, but that's actually a *directory* (not a file!)...

I think that should say it's named `dictionary.txt`.

It also says:

> This dictionary is a simple word list in plain-text. You can add and remove words at will.

... which is ... nice, but doesn't say clearly what the format is: is it one word per line? or something else?

It *looks* like one word per line, but perhaps that could be made clearer...

> You can add to the user dictionary with code actions on misspelled words.

For what it's worth, that's the key to the whole thing, and thought me a new thing about LSP and eglot i didn't know (cf #150): <kbd>C-c e a</kbd> triggers the "code action" thing, which then prompts you for things...

Adding to the dictionary takes a long time though (a few seconds), not sure what's up with that... 

It also doesn't work: either `harper-ls` doesn't reload the dict on the fly, or it's reading from a different path that it's writing to, because picking "add to global" as an action doesn't actually fix the issue in LSP.

I should finally point out the last line in that README file is a bit confusing to me as well:

> I do take pull requests or issues for adding words to the static dictionary. It is composed of two files: `harper-core/dictionary.dict` and `harper-core/dictionary.dict`

... isn't that the same file twice? :)

**Comment by elijah-potter** [comment]
> ... which is ... nice, but doesn't say clearly what the format is: is it one word per line? or something else?

That's exactly right. I'll add that clarification to the docs.

 > ... isn't that the same file twice? :)

Yup. That's a mistake. I've fixed that as well.

If you just want to add some proper nouns (Grafana, Prometheus), run `just addnoun <your noun>` and open a PR.

> Adding to the dictionary takes a long time though (a few seconds), not sure what's up with that...

That's pretty concerning. Are you still having this issue? Are any logs getting printed?

**Reply by anarcat** [reply]
nice, but i still don't see the dict being clearly stated as being in `dictionary.txt`, perhaps i should just make a PR instead here. :)

> If you just want to add some proper nouns (Grafana, Prometheus), run `just addnoun <your noun>` and open a PR.

neat! for the record, here is my current dictionary:

```
CI
RRD
TCP
DSL
TPA
CRM
VPN
SLA
FPM
IMAP
Munin
IPsec
Golang
GitLab
Nagios
Icinga
PromQL
Joomla
Grafana
anarcat
systemd
CiviCRM
FastCGI
Postfix
sharding
natively
journald
Wordpress
Javascript
Pushgateway
Alertmanager
StackExchange
```

not sure if that's all relevant for public consumption. :)

> That's pretty concerning. Are you still having this issue? Are any logs getting printed?

I'll try to keep an eye on my logs and keep you informed, thank you so much for the prompt response!

**Reply by anarcat** [reply]
> I'll try to keep an eye on my logs and keep you informed, thank you so much for the prompt response!

I moved that to #185.

**Reply by elijah-potter** [reply]
I like a lot of those. I'll add 'em!

**Reply by anarcat** [reply]
i actually filed https://github.com/elijah-potter/harper/issues/197 to followup on those words!

**Comment by elijah-potter** [comment, marked as answer]
If someone sees this discussion looking for documentation on the dictionary, [take a look at this page](https://writewithharper.com/docs/contributors/dictionary). If there's any information missing, don't be afraid to ask.

**Reply by faervan** [reply]
and look at [this page](https://writewithharper.com/docs/integrations/language-server#Dictionaries) too :)
